api/observability/v1/clusterlogforwarder_types.go

```diff
@@ -0,0 +1,190 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1
+
+import (
+	"github.com/openshift/cluster-logging-operator/internal/status"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterLogForwarderSpec defines the desired state of ClusterLogForwarder
+type ClusterLogForwarderSpec struct {
+	// Inputs are named filters for log messages to be forwarded.
+	//
+	// There are three built-in inputs named `application`, `infrastructure` and
+	// `audit`. You don't need to define inputs here if those are sufficient for
+	// your needs. See `inputRefs` for more.
+	//
+	// +optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Forwarder Inputs",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:forwarderInputs"}
+	Inputs []InputSpec `json:"inputs,omitempty"`
+
+	// Outputs are named destinations for log messages.
+	//
+	// There is a built-in output named `default` which forwards to the default
+	// openshift log store. You can define outputs to forward to other stores or
+	// log processors, inside or outside the cluster.
+	//
+	// +optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Forwarder Outputs",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:forwarderOutputs"}
+	Outputs []OutputSpec `json:"outputs,omitempty"`
+
+	// Filters are applied to log records passing through a pipeline.
+	// There are different types of filter that can select and modify log records in different ways.
+	// See [FilterTypeSpec] for a list of filter types.
+	// +optional
+	Filters []FilterSpec `json:"filters,omitempty"`
+
+	// Pipelines forward the messages selected by a set of inputs to a set of outputs.
+	//
+	// +required
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Forwarder Pipelines",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:forwarderPipelines"}
+	Pipelines []PipelineSpec `json:"pipelines,omitempty"`
+
+	// ServiceAccount is the spec of the serviceaccount associated with the clusterlogforwarder
+	// +required
+	ServiceAccountName ServiceAccountSpec `json:"serviceAccount,omitempty"`
```

Comment:
For same-namespace resources we can simply use k8s structs like `LocalObjectReference` which are each to update to cross-namespace by switching to `ObjectReference`:
- https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/local-object-reference/
- https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-reference/

Also another option is to use `TypedLocalObjectReference` to allow referencing only `ServiceAccount` as a resource.

IIRC, also it guards for naming conventions on `names` as such: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names